src/main/java/Controller/ProductComparatorController.java

```diff
@@ -0,0 +1,111 @@
+package Controller;
+
+import Entities.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+
+import UseCases.Comparator.ProductComparator;
+import UseCases.Comparator.ProductComparatorFactory;
+
+/**
+ * a controller class that uses comparator factory to create the different types of comparators
+ * and sorts the wishlist using that comparator
+ */
+public class ProductComparatorController implements ProductList {
```

Comment:
Why does this implement product list and have add and remove item functions? This is a controller just for sorting is it not?